### PR TITLE
Display context menus in webviews for links, images and any content.

### LINKED ISSUE #347: Context Menu: Links
Tracking for MVP milestone.
Should not be considered complete without tests.

Should do the following:
* Open in new tab
* Open in new Window
* Copy link address
* Copy 
* Select All
* Inspect Element

Feel free to split up in multiple issues if you're working on this.

### PR DESCRIPTION
**Pull request by jsantell**
Fixes #347. r=vporof

Ideally, we wouldn't need a thunk in the `displayWebviewContextMenu` action creator, but we need dispatch in the saga, since I can't think of a reasonable way to dispatch via yield/put/call in the bound handlers without the saga terminating (and any way I can think of involves needing a 'close' event on the menu which does not exist AFAICT)

### CODE REVIEW

File: app/ui/browser-modern/actions/page-effects.js [hunk 1]
```diff
@@ -248,3 +248,12 @@ export function getCertificateError(pageId, url) {
     url,
   };
 }
+
+export function displayWebviewContextMenu(webview, e) {
+  return dispatch => dispatch({
+    type: EffectTypes.DISPLAY_WEBVIEW_CONTEXT_MENU,
+    webview,
+    e,
+    dispatch,
```
Comment (victorporof) [inline app/ui/browser-modern/actions/page-effects.js]: Don't use dispatch in sagas. Use `put` instead.

File: app/ui/browser-modern/sagas/page-sagas.js [hunk 2]
```diff
@@ -170,3 +177,54 @@ function* getCertificateError({ pageId, url }) {
...
+  menu.append(new MenuItem({
+    label: 'Select All',
+    click: () => webview.selectAll(),
+  }));
+
+  menu.append(new MenuItem({ type: 'separator' }));
+
+  menu.append(new MenuItem({
+    label: 'Inspect Element',
+    click: () => webview.inspectElement(e.x, e.y),
```
Comment (victorporof) [inline app/ui/browser-modern/sagas/page-sagas.js]: This should dispatch a `TOGGLE_DEVTOOLS` effect instead.

File: app/ui/browser-modern/views/browser/page/index.jsx [hunk 3]
```diff
@@ -41,6 +41,7 @@ class Page extends Component {
   constructor(props) {
     super(props);
     this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+    this.requestContextData = this.requestContextData.bind(this);
```
Comment (victorporof) [inline app/ui/browser-modern/views/browser/page/index.jsx]: Please use autobinding methods instead.

File: app/ui/browser-modern/views/browser/page/index.jsx [hunk 4]
```diff
@@ -226,6 +230,11 @@ class Page extends Component {
     this.webview.dispatchEvent(ev);
   }
 
+  requestContextData(e) {
```
Comment (victorporof) [inline app/ui/browser-modern/views/browser/page/index.jsx]: requestContextData = (e) => { ... }

**Comment by victorporof** [comment]
Here's a patch removing `dispatch`: https://pastebin.mozilla.org/8916539

**Comment by jsantell** [comment]
Me and @victorporof discussed the strategy of using `yield put` and `takeLatest` to only have one saga running at a time, even though it will never 'complete' if the user selects no context menu option

**Pull request merged**